monkey/agent_plugins/payloads/cryptojacker/src/plugin.py

```diff
@@ -48,6 +51,7 @@ def run(
                 agent_id=self._agent_id,
                 agent_event_publisher=self._agent_event_publisher,
                 island_server_address=self._island_server_address,
+                local_machine_info=self._local_machine_info,
```

Comment:
I considered that but (1) it makes the plugin framework inconsistent, (2) local machine info is basic information that may have other useful things that the payload could use in the future.

Reply:
Also, ATM, everything that needs local machine info uses just the OS except Log4Shell.

Reply:
> (1) it makes the plugin framework inconsistent

I don't understand what you mean. `plugin.py` implements the plugin interface. All plugin.py files have methods that accept `LocalMachineInfo`

The role of `plugin.py` is ultimately to preserve the interface segregation principle, "no code should be forced to depend on methods it does not use." The `plugin.py` transforms the raw imports that all plugins are passed into the inputs that are most useful for the code that implements the business logic. For example, PowerShell's `plugin.py` doesn't pass the OS to the underlying code. It instead passes "running_on_windows", because that's what is valuable.

Reply:
My bad, I was thinking this was the plugin constructor.